File: spec/euler/level1/problem001_spec.clj
(ns euler.level1.problem001-spec
    (:use
      [speclj.core]
      [euler.level1.problem001]))

; http://projecteuler.net/index.php?section=problems&id=1

(describe "Euler Problem #1"

  (it "divisible by 3 or 5?"
    (should= true (divisible-by-3-or-5? 0))
    (should= false (divisible-by-3-or-5? 1))
    (should= false (divisible-by-3-or-5? 2))
    (should= true (divisible-by-3-or-5? 3))
    (should= false (divisible-by-3-or-5? 4))
    (should= true (divisible-by-3-or-5? 5))
    (should= true (divisible-by-3-or-5? 6))
    (should= false (divisible-by-3-or-5? 7))
    (should= false (divisible-by-3-or-5? 8))
    (should= true (divisible-by-3-or-5? 9))
    (should= true (divisible-by-3-or-5? 10)))

  (it "finds desired numbers less than n"
    (should= [] (desired-numbers-less-than 0))
    (should= [3] (desired-numbers-less-than 4))
    (should= [3] (desired-numbers-less-than 5))
    (should= [3 5] (desired-numbers-less-than 6))
    (should= [3 5 6 9] (desired-numbers-less-than 10)))

  (it "Solves #1"
    (should= 23 (euler-1 10))
    (should= 233168 (euler-1 1000))))

File: ruby/spec/problem001_spec.rb
require 'problem001'

# https://projecteuler.net/index.php?section=problems&id=1

describe "Euler Problem #1" do
  it "Solves #1" do
    expect(euler_1(1)).to eq 0
    expect(euler_1(2)).to eq 0
    expect(euler_1(3)).to eq 0
    expect(euler_1(4)).to eq 3
    expect(euler_1(5)).to eq 3
    expect(euler_1(6)).to eq 8
    expect(euler_1(7)).to eq 14
    expect(euler_1(8)).to eq 14
    expect(euler_1(9)).to eq 14
    expect(euler_1(10)).to eq 23
    expect(euler_1(1000)).to eq 233168
  end
end
